src/components/TemplateList/TemplateList.js

```diff
@@ -259,13 +259,27 @@ const TemplateList = ({
           let file = url;
           let headers = new Headers();
           headers.append('Authorization', bearerToken);
-          const fileName = `${studyKfid ? studyKfid : 'data'}_templates${
+          var fileName = `${studyKfid ? studyKfid : 'data'}_templates${
             format === 'zip' ? '.zip' : '.xlsx'
           }`;
           fetch(file, {headers})
-            .then(response => response.blob())
+            .then(response => {
+              fileName = response.headers
+                .get('Content-Disposition')
+                .split('filename*=UTF-8')[1]
+                .substring(2);
+              response.blob();
+            })
             .then(blobby => {
-              let objectUrl = window.URL.createObjectURL(blobby);
+              var binaryData = [];
+              binaryData.push(blobby);
+              let objectUrl = window.URL.createObjectURL(
+                new Blob(binaryData, {
+                  type: fileName.endsWith('zip')
```

Comment:
Tried it and I'm getting the issue I had before, my machine isn't able to open any of the downloads bc it says the file is in an unsupported format. I see this for both zip and xlsx 

Btw, you should probable set this type directly from the response's `Content-Type` header